django/db/models/base.py

```diff
@@ -244,13 +244,21 @@ def __new__(cls, name, bases, attrs):
                     field = None
                 new_class._meta.parents[base] = field
             else:
+                base_parents = base._meta.parents.copy()
+
                 # .. and abstract ones.
                 for field in parent_fields:
                     new_field = copy.deepcopy(field)
                     new_class.add_to_class(field.name, new_field)
+                    # Replace parent links defined on this base by the new
+                    # field as they will be appropriately resolved if required.
+                    if isinstance(field, OneToOneField):
```

Comment:
I guess we can change this check as we don't deal with any of `OneToOneField` internals.